sklearn/preprocessing/imputation.py

```diff
@@ -372,4 +409,15 @@ def transform(self, X):
 
             X[coordinates] = values
 
+            if self.add_missing_indicator:
+                features_with_missing_values = np.where(np.any
+                                                        (mask, axis=0))[0]
+                imputed_mask = mask[:, features_with_missing_values]
+                X = np.hstack((X, imputed_mask))
+                if self.axis == 0:
+                    self.imputed_features_ = valid_idx
+                    [features_with_missing_values]
```

Comment:
Thanks, I just realised what this meant. Will never forget !
Shall I make separate functions like ``_sparse_transform`` and ``_dense_transform`` on the lines of ``fit`` then ?